django/core/serializers/base.py

```diff
@@ -282,9 +282,14 @@ def m2m_convert(value):
         def m2m_convert(v):
             return model._meta.pk.to_python(v)
 
+    try:
+        pks_iter = iter(field_value)
+    except TypeError as e:
+        raise M2MDeserializationError(e, field_value)
```

Comment:
I'm not familiar with the whole (de)serializer codebase but it seems to me that when we reach this function then `field_value` should really be iterable.

If we're trying to deserialize `None` or anything un-iterable then something has gone wrong, no? Wouldn't it be better to error out rather than try to correct the issue?

Reply:
Agreed.